janitor/functions/concatenate_columns.py

```diff
@@ -52,7 +52,10 @@ def concatenate_columns(
         raise JanitorError("At least two columns must be specified")
 
```

Comment:
```suggestion

    df = df.copy()  # avoid mutating original data
```

I note this function also mutates the original `df`. This is a bug.
Please update this into changelog.

```python
>>> df = pd.DataFrame([[1, 2], [3, None]], columns=['a', 'b'])
>>> df.concatenate_columns(
...     column_names=["a", "b"], new_column_name="m",
... )
   a    b      m
0  1  2.0  1-2.0
1  3  NaN  3-nan
>>> df  # changed
   a    b      m
0  1  2.0  1-2.0
1  3  NaN  3-nan
```